ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/passthrough/PassThroughHandler.java

```diff
@@ -43,27 +50,40 @@ public void handle(final RoutingContext context, final JsonRpcRequest request) {
         ethNodeClient.request(
             httpServerRequest.method(),
             httpServerRequest.uri(),
-            proxiedResponse -> {
-              logResponse(proxiedResponse);
-
-              httpServerRequest.response().setStatusCode(proxiedResponse.statusCode());
-              httpServerRequest.response().headers().setAll(proxiedResponse.headers());
-              httpServerRequest.response().setChunked(false);
-
-              proxiedResponse.bodyHandler(
-                  data -> {
-                    logResponseBody(data);
-                    // End the sendRequest, preventing any other handler from executing
-                    httpServerRequest.response().end(data);
-                  });
-            });
+            response -> handleResponse(context, response));
 
+    proxyRequest.setTimeout(httpRequestTimeout.toMillis());
+    proxyRequest.exceptionHandler(thrown -> requestExceptionHandler(context, thrown));
     proxyRequest.headers().setAll(httpServerRequest.headers());
     proxyRequest.setChunked(false);
     proxyRequest.end(context.getBody());
     logRequest(request, httpServerRequest);
   }
 
+  private void requestExceptionHandler(final RoutingContext context, final Throwable thrown) {
```

Comment:
could just call fail with the exception and get the jsonRpcErrorHandler to deal with it?

Reply:
THAT is probably nicer.

Reply:
Decided not to do this, as we are able to indicate that that the failure was either an internal error or a gateway timeout.